scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,567 @@
+
+
+# Exit on error
+set -e
+
+
+# Global variables
+timestamp="$(date +%F_%T)"
+output_folder="/tmp/kubestellar-snapshot"
+
+
+# Script info
+SCRIPT_NAME="KubeStellar Snapshot"
+SCRIPT_VERSION="0.10"
+
+
+# Colors
+COLOR_NONE="\033[0m"
+COLOR_RED="\033[1;31m"
+COLOR_GREEN="\033[1;32m"
+COLOR_BLUE="\033[94m"
+COLOR_YELLOW="\033[1;33m"
+COLOR_PURPLE="\033[1;35m"
+COLOR_ERROR="${COLOR_RED}"
+COLOR_STATUS_TRUE="${COLOR_GREEN}"
+COLOR_STATUS_FALSE="${COLOR_RED}"
+COLOR_INFO="${COLOR_BLUE}"
+COLOR_TITLE="${COLOR_YELLOW}"
+COLOR_YAML="${COLOR_PURPLE}"
+
+
+# Command line arguments
+arg_kubeconfig=""
+arg_context=""
+arg_logs=false
+arg_yaml=false
+arg_verbose=false
+
+
+# Display the command line help
+display_help() {
+  cat << EOF
+Usage: $0 [--kubeconfig|-K <filename>] [--context|-C <name>] [--logs|-L] [--yaml|-Y] [--verbose|-V] [--version|-v] [--help|-h] [-X]
+
+--kubeconfig|-K <filename> use the specified kubeconfig to find KubeStellar Helm chart
+--context|-C <name>        use the specified context to find KubeStellar Helm chart
+--logs|-L                  save the logs of the pods
+--yaml|-Y                  save the YAML of the resources
+--verbose|-V               output extra information
+--version|-v               print out the script version
+--help|-h                  show this information
+-X                         enable verbose execution for debugging
+
+Note: After piping the output of the script to file use "more"
+      to inspect the file content in color. ALternatively install
+      "ansi2txt" and pipe the script through it to remove escape
+      characters and obtain a plan text report.
+EOF
+}
+
+
+# Indent JSON
+indent() {
+    sed 's/^/  /'
+}
+
+# Echo in color
+echocolor() {
+    # $1 = color
+    # $2 = message
+    echo -e "$1$2${COLOR_NONE}"
+}
+
+
+# Echo to stderr
+echoerr() {
+    # $1 = error message
+    >&2 echocolor ${COLOR_ERROR} "ERROR: $1"
+}
+
+
+# Echo colorized title
+echotitle() {
+    # $1 = message
+    echocolor ${COLOR_TITLE} "\n$1"
+}
+
+
+# Echo the status in color
+echostatus() {
+    # $1 = status text
+    status="$(echo $1 | tr '[:upper:]' '[:lower:]')" # lowercase
+    if [[ "true succeeded running active 1" =~  "$status" ]] ; then
+        echocolor ${COLOR_STATUS_TRUE} "$status"
+    else
+        echocolor ${COLOR_STATUS_FALSE} "$status"
+    fi
+}
+
+
+# Check if a pre-requisite is installed
+is_installed() {
+    # $1 == name
+    # $2 == command name to search
+    # $3 == command to get the version, unstructured
+    if which $2 > /dev/null ; then
+        echov -e "${COLOR_GREEN}\xE2\x9C\x94${COLOR_NONE} ${COLOR_INFO}$1${COLOR_NONE} version ${COLOR_INFO}$(eval "$3" 2> /dev/null || true)${COLOR_NONE} at ${COLOR_INFO}$(which $2)${COLOR_NONE}"
+    else
+        echoerr "missing dependency:"
+        echo -e "${COLOR_RED}X${COLOR_NONE} $1"
+        exit 1
+    fi
+}
+
+
+# Get the kubeconfig of a particular Control Plane
+get_kubeconfig() {
+    context="$1" # context in the kubeconfig
```

Comment:
Context in the _host_ kubeconfig. Worth clarifying _which_ kubeconfig because there are multiple in sight.

Reply:
I changed to `context="$1" # context in the host kubeconfig or the config passed to the script using --kubeconfig flag`

Reply:
Not a big deal, but: I would not write "host kubeconfig" --- because, as an adjective, "host" (or, as I prefer, "hosting" or "KubeFlex hosting") applies to a context rather than a kubeconfig. For your purpose here I would simply write "relevant kubeconfig" or "applicable kubeconfig" or "chosen kubeconfig".